### PR TITLE
Tagging using RAKE and spaCy

### PR DESCRIPTION
**Pull request by arjun-vasudevan**
Rake

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,47 @@
-APScheduler==3.7.0
+appdirs==1.4.4
```
Comment (susheelk) [inline requirements.txt]: Damn do we need all these dependencies?

File: feed_parsers/implemented_feeds.py [hunk 2]
```diff
@@ -12,11 +13,17 @@ def _parse_last_updated(self, feed: feedparser) -> datetime:
     def _parse_articles(self, feed: feedparser) -> [NewsArticle]:
         articles = []
         for e in feed.entries:
+            desc_title = regex.search('(?<=title=[\'"])([\s\w.,:!?\\\/-]*)', e.description)
+            desc_title = desc_title.group(0) if desc_title else ''
+
+            desc_para = regex.search('(?<=<p>)(.*?)(?=<\/p>)', e.description).group(0)
```
Comment (susheelk) [inline feed_parsers/implemented_feeds.py:19]: nice regex lol

File: requirements.txt (hunk 1 above)
Comment (arjun-vasudevan) [reply requirements.txt]: Yep

File: pipeline/tagger.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+from pipeline.pipeline_stage import PipelineStage
+
+
+class Tagger(PipelineStage):
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = spacy.load("en_core_web_md")
+        self.pos_tag = ['PROPN', 'NOUN']
```
Comment (susheelk) [inline pipeline/tagger.py]: (nit) you could make this a static variable

File: pipeline/tagger.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            if token.pos_ in self.pos_tag:
+                result.append(token.text)
+
+        spacys = [x[0] for x in Counter(result).most_common()]
+
+        to_process.tags = list(set(rakes + spacys))
+
+        return to_process
+
+    def start(self):
```
Comment (susheelk) [inline pipeline/tagger.py:52]: You can remove this method or raise `NotImplementedError`

**Review by susheelk — CHANGES_REQUESTED** [review]
# +1
Resolve the comments and merge at your discretion :+1:

**Pull request merged**